src/cluster_legacy.c

```diff
@@ -6517,3 +6566,31 @@ int clusterAllowFailoverCmd(client *c) {
 void clusterPromoteSelfToMaster(void) {
     replicationUnsetMaster();
 }
+
+void updateNodesHealth(void) {
+    dictIterator *di;
+    dictEntry *de;
+    clusterNode *node;
+    int overall_health_changed = 0;
+    di = dictGetSafeIterator(server.cluster->nodes);
+    while((de = dictNext(di)) != NULL) {
+        node = dictGetVal(de);
+        int present_node_health;
+
+        long long node_offset = getNodeOffSet(node);
+
+        if (nodeFailed(node) || (nodeIsSlave(node) && node_offset == 0)) {
+            present_node_health = 0;
+        } else {
+            present_node_health = 1;
+        }
```

Comment:
```suggestion
        int present_node_health = isReplicaAvailable(node);
```
I think this is just this right? We may want to rename `isReplicaAvailable` to something a bit more generic, we may also need to export this function.

Reply:
How about `isNodeHealthy`?

Reply:
If we move this function to cluster.c, we don't need to export it.

Reply:
Yeah the split between these files is weird. I wonder if we can move the node structs like `clusterNode` to cluster.c and keep the bus protocol like clusterMsg in cluster_legacy.c. That might be a better split.